LandSandBoat/server
# [Jobs][DRK] Create job_utils/dark_knight, update Arcane Circle, Arcane Crest, Blood Weapon

**_I affirm:_**

## What does this pull request do?



**Job Abilities**

- Create job_utils/dark_knight.lua and populates necessary ability check and ability use functions.
- Moves functionality from all Dark Knight globals/abilities files to globals/job_utils/dark_knight.lua.
- Adjusts Arcane Circle duration handling and adds consideration for new ARCANE_CIRCLE_POTENCY mod.
- Adds TODO and comments for #2849.
- Sets up Arcane Crest to be usable against Arcana enemies only.
- Sets up Blood Weapon to add duration from the Enhances "Blood Weapon" effect mod.

**Effects**

- Sets up the Arcane Crest status effect and debuffs.

**Mods**

- Converts ARCANE_CIRCLE_DURATION from a 90s increase to a 50% increase on Chaos/Ignominy Sollerets.
- Creates ARCANE_CIRCLE_POTENCY and adds it to ability base strength on Chaos/Ignominy Sollerets.
- Creates ENHANCES_BLOOD_WEAPON and adds it to Blood Weapon base duration.

**Database**

- Adjusts ARCANE_CIRCLE_DURATION from +duration to * %duration on Chaos/Ignominy Sollerets in item_mods.sql.
- Adds ARCANE_CIRCLE_POTENCY to Chaos/Ignominy Sollerets in item_mods.sql.

## References

Dark Knight - https://www.bg-wiki.com/ffxi/Dark_Knight
Arcane Circle - https://www.bg-wiki.com/ffxi/Arcane_Circle
Arcane Crest - https://www.bg-wiki.com/ffxi/Arcane_Crest
Blood Weapon - https://www.bg-wiki.com/ffxi/Blood_Weapon

## Steps to test these changes



1. Upload changes to your server.
2. Load the server.
3. Use the `!changejob DRK 99` command.
4. Use all JA's to confirm they do not throw errors.
5. Use the !setmod command to add the updated ability modifiers.
